apps/api/v2/src/modules/auth/strategies/api-auth/api-auth.strategy.ts

```diff
@@ -20,19 +22,74 @@ export class ApiAuthStrategy extends PassportStrategy(BaseStrategy, "api-auth")
     private readonly oauthFlowService: OAuthFlowService,
     private readonly tokensRepository: TokensRepository,
     private readonly userRepository: UsersRepository,
-    private readonly apiKeyRepository: ApiKeyRepository
+    private readonly apiKeyRepository: ApiKeyRepository,
+    private readonly oauthRepository: OAuthClientRepository,
+    private readonly profilesRepository: ProfilesRepository
   ) {
     super();
   }
 
   async authenticate(request: Request) {
-    const authString = request.get("Authorization")?.replace("Bearer ", "");
-    if (!authString) {
-      return this.error(new UnauthorizedException("No Authorization header provided"));
+    try {
+      const { params } = request;
+      const oAuthClientSecret = request.get(X_CAL_SECRET_KEY);
+      const oAuthClientId = params.clientId || request.get(X_CAL_CLIENT_ID);
+      const bearerToken = request.get("Authorization")?.replace("Bearer ", "");
+
+      if (oAuthClientId && oAuthClientSecret) {
+        return await this.authenticateOAuthClient(oAuthClientId, oAuthClientSecret);
+      }
+
+      if (bearerToken) {
+        const requestOrigin = request.get("Origin");
+        return await this.authenticateApiKey(bearerToken, requestOrigin);
+      }
```

Comment:
we should rename this, I was wondering where was the access token auth happening until I realised it's inside authenticateApiKey. maybe rename to authenticateBearerToken

Reply:
Implemented in [refactor: rename authenticateApiKey to authenticateBearerToken](https://github.com/calcom/cal.com/pull/15928/commits/2af11ac671d171923e6deb0b720f31fb33bb52bb)